src/webgpu/api/validation/capability_checks/limits/maxColorAttachmentBytesPerSample.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { range } from '../../../../../common/util/util.js';
+import { roundDown } from '../../../../util/math.js';
+
+import {
+  kLimitBaseParams,
+  kLimitModes,
+  LimitTestsImpl,
+  makeLimitTestGroup,
+} from './limit_utils.js';
+
+function getPipelineDescriptor(
+  device: GPUDevice,
+  actualLimit: number,
+  testValue: number
+): GPURenderPipelineDescriptor | undefined {
+  const numRGBA32F = actualLimit / 16;
+  const numR8Unorm = testValue - actualLimit;
+  const maxColorAttachments = device.limits.maxColorAttachments;
+  if (numRGBA32F + numR8Unorm > maxColorAttachments) {
+    return;
+  }
+
+  const rgba32fTargets = new Array(numRGBA32F).fill({
+    format: 'rgba32float',
+    writeMask: 0,
+  }) as GPUColorTargetState[];
+
+  const r8UnormTargets = new Array(numR8Unorm).fill({
+    format: 'r8unorm',
+    writeMask: 0,
+  });
+
+  const code = `
+    @vertex fn vs() -> @builtin(position) vec4f {
+      return vec4f(0);
+    }
+
+    @fragment fn fs() -> @location(0) vec4f {
+      return vec4f(0);
+    }
+  `;
+  const module = device.createShaderModule({ code });
+  return {
+    layout: 'auto',
+    vertex: {
+      module,
+      entryPoint: 'vs',
+    },
+    fragment: {
+      module,
+      entryPoint: 'fs',
+      targets: [...rgba32fTargets, ...r8UnormTargets],
+    },
+  };
+}
+
+function createTextures(t: LimitTestsImpl, format: GPUTextureFormat, numTextures: number) {
+  return range(numTextures, _ =>
+    t.trackForCleanup(
+      t.device.createTexture({
+        size: [1, 1],
+        format,
+        usage: GPUTextureUsage.RENDER_ATTACHMENT,
+      })
+    )
+  );
+}
+
+const limit = 'maxColorAttachmentBytesPerSample';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createRenderPipeline,at_over')
```

Comment:
can we additionally test:
 - multisampled formats - because the limit is "perSample", not "perTexel"
  
 - formats or series of formats where `renderTargetComponentAlignment` is important. like when you're at a 1-byte offset but the next format requires 4-byte alignment.

Reply:
added